src/coap_debug.c

```diff
@@ -189,37 +197,44 @@ coap_print_addr(const struct coap_address_t *addr, unsigned char *buf, size_t le
 
     break;
   default:
-    memcpy(buf, "(unknown address type)", min(22, len));
+    /* Include trailing NULL if possible */
+    memcpy(buf, "(unknown address type)", min(22+1, len));
+    buf[len-1] = '\000';
     return min(22, len);
   }
 
   /* Cast needed for Windows, since it doesn't have the correct API signature. */
   if (inet_ntop(addr->addr.sa.sa_family, addrptr, (char *)p,
                 min(len, need_buf)) == 0) {
     perror("coap_print_addr");
+    *p = '\000';
     return 0;
   }
 
-  p += strnlen((char *)p, len);
+  p += strlen((char *)p);
```

Comment:
Unfortunately even RFC2553 is not clear.
````
The inet_ntop() function converts a numeric address into a text
string suitable for presentation.
````
So no total clarity.  Everywhere refers to the buffer size must be a defined minimum size, so will change the code so that if there is insufficient provided space, nothing is filled in in the return from `coap_print_addr()`.